### PR TITLE
Setup Hardhat in CI

### PR DESCRIPTION
**Pull request by crabsinger**
## Setup Hardhat in CI

- setup running of hardhat tests in CI
- ~~only run if changes to hardhat/ subdir~~ cypress fails with conditional running WTF?
- switch to using yarn link
- rebuild dist/ and check in latest changes

github actions - which is only used for code coverage, is failing on one hardhat test...

### CODE REVIEW

**Comment by crabsinger** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2146** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/2146" target="_blank">Graphite</a>  👈
    * **PR #2141** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/2141" target="_blank">Graphite</a> 
      * **PR #2143** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/2143" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/coordinape/coordinape/2146?utm_source=stack-comment).

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -36,6 +36,7 @@ jobs:
 
     steps:
       - uses: actions/checkout@v3.1.0
+      - run: git submodule sync && git submodule update --init
```
Comment (levity) [inline .github/workflows/tests.yml:39]: should this still be here? @crabsinger

File: scripts/env.sh [hunk 2]
```diff
@@ -36,8 +36,10 @@ if [ "$SET_CI_VARS" ]; then
   export NODE_ENV=development
   export LOCAL_LOCALSTACK_PORT_RANGE="4666-4683"
   export LOCAL_HASURA_PORT=8087
+  export LOCAL_WEB_PORT=3007
   export LOCAL_POSTGRES_PORT=5437
   export LOCAL_WEB_PORT=3007
+  export LOCAL_SEED_ADDRESS=0x21DF875603Ba765a15F9ae20e633D2DAe42952d9
```
Comment (levity) [inline scripts/env.sh:42]: is this still needed?

**Review by levity — COMMENTED** [review]
See comments

File: scripts/env.sh (hunk 2 above)
Comment (crabsinger) [reply scripts/env.sh:42]: @levity yes, this is needed. this is the fix for the bug in Circle CI's path filtering orb which interprets a string "0x21DF875603Ba765a15F9ae20e633D2DAe42952d9" as a number and breaks everything.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (crabsinger) [reply .github/workflows/tests.yml]: removed

**Review by levity — APPROVED** [review]

**Pull request merged**